### ISSUE TITLE
Missing information for Job Table

### ISSUE THREAD

**Issue by ag-m2ms** [m2ms/fragalysis-frontend#878]
For the job table in #865 we are missing some data to add remaining columns. The columns are:

- Job class - @rsanchezgarc was supposed to answer what this one might be
- Job type - again we are not sure what this one might hold, we already display `squonk_job_name` in the `name` column
- Date - When the job was triggered

We use `/api/job_request/` endpoint to populate the table.

**Comment by rsanchezgarc** [comment]
Job class would be a string like "docking", "scoring", etc. We still need to define an ontology and this particular type of info should be included in the job definition. For the moment, you can just display N/A if one of the fields is not available.

Job type should be squonk_job_name and name could be a user-provided alias for the particular job.

**Comment by alanbchristie** [comment]
Where is the source of **job class**? Is it, for example, a "keyword" or category" that the developer of a **Job** would place in the Job's original definition?

With regard to **date** I will add two useful fields that will be exposed via `/api/job_request/`. These will be ISO8601-formatted strings identifying the date/time the Job started (`job_start_time`) and another that identifies when the Job (including any automated upload) completed (`job_finish_time`).

**Comment by rsanchezgarc** [comment]
job class should be a category. ATM we haven't defined the possible categories, but at least we will have: pose prediction (docking), pose scoring, compound scoring, pose-refinement, fragment-elaboration-growth, fragment-elaboration-merge. I guess it should be placed in the job definition by the developer

**Comment by alanbchristie** [comment]
The development effort for #859 introduces new fields in the `JobRequest` table that should satisfy the backend effort for this issue. For...

1. **Job CLASS** - this will be the responsibility of the Job Definition author who will populate the _Squonk JobDefinition -> category_ with a suitable string (see below)
2. **Job TYPE** - this is the name assigned to the job in the Job Definition (i.e. `assemble-conformers`) 
3. **DATE** - when the job was triggered. This is available in the `JobRequest -> job_start_datetime`

Classes (categories) known here are: -

- `pose prediction`
- `pose scoring`
- `compound scoring`
- `pose refinement`
- `fragment elaboration growth`
- `fragment elaboration merge`

When comparing classes case (upper/lower) should be disregarded, i.e. "Pose Scoring" should be considered the same as "pose scoring"

---

Actions: -

a) Populate JobDefinition category strings
b) Adjust F/E to display the CLASS and TYPE using the JobDefinition YAML and the JobRequest to obtain Job start time.

**Comment by rsanchezgarc** [comment]
@alanbchristie We also need 'fragment elaboration link'

**Comment by tdudgeon** [comment]
The job category us defined in the `jobs.<job-name>.category` field int he job definition YAML (for example see https://github.com/InformaticsMatters/virtual-screening/blob/main/data-manager/im-virtual-screening.yaml#L49).
Ruben and I agreed that it would make sense to add an optional `subcategory` field, but no further sub-division.
 
I've defined a google sheet where we can go over this categorisation of jobs:
https://docs.google.com/spreadsheets/d/1RwbkDfOoY9KuCFQicJ73wYAMupXV6b44waI_GXJFOIE/edit#gid=0

It lists the CURRENT definitions, not necessarily the ones we want to end up with.

**Comment by tdudgeon** [comment]
Regarding the comments @boriskovar-m2ms had about the current content of the jobs table, here is some context.

The job really has two names. The job definition has a name (e.g. `fragmenstein-combine`) but when the job is launched in Squonk the user can give that job a specific name. The default is the job definition name, but the user can edit this if they want to. Fragalysis does not currently seem to allow the user to specify this specific name, but probably should do. If this happens both names could potentially be displayed in the jobs table, or possibly just the specific name.

The "Show in Squonk" button seems like a good idea, but bear in mind that the job may have subsequently been deleted in Squonk.

And on the topic of deletion we will want to add options to delete the job. This has a number of aspects:
1. Delete the job in Squonk
2. Also delete job's results in Squonk
3. Archive the job in Squonk so that it won't get automatically deleted.
4. Delete the job from Fragalysis.
5. Also delete the job's results in Fragalysis (so that it no longer appears as a computed set)

**Comment by boriskovar-m2ms** [comment]
@tdudgeon  yes I know. We just need to add a name field to the job launcher dialog and that is. Now we are just passing hardcoded value 'fragmenstein-combine' and I wanted to change that. No BE work involved.

**Comment by rsanchezgarc** [comment]
@boriskovar-m2ms Is this solved?
